pizza_cutter/metadetect/run_metadetect.py

```diff
@@ -69,10 +70,15 @@ def _post_process_results(*, outputs, obj_data, image_info):
 def _do_metadetect(config, mbobs, seed, i, preprocessing_function):
     _t0 = time.time()
     rng = np.random.RandomState(seed=seed)
-    if preprocessing_function is not None:
-        logger.debug("preprocessing multiband obslist %d", i)
-        mbobs = preprocessing_function(mbobs=mbobs, rng=rng)
-    res = do_metadetect(config, mbobs, rng)
+
+    n_obs = sum(len(obsl) for obsl in mbobs)
```

Comment:
we need all to be > 0, we don't want to do fits without some band

Reply:
see my solution in the other

Reply:
yup - this is why yours is better!